pallets/subbridge/src/xcmbridge.rs

```diff
@@ -275,17 +275,18 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		// Return dest chain and beneficiary on dest chain
 		pub fn extract_dest(dest: &MultiLocation) -> Option<(MultiLocation, MultiLocation)> {
-			if dest.is_here() {
-				None
-			} else {
-				match dest.last() {
-					Some(_) => {
-						let mut dest_location = dest.clone();
-						let beneficiary = dest_location.take_last().unwrap().into();
-						Some((dest_location, beneficiary))
-					}
-					_ => None,
-				}
+			match (dest.parents, dest.first_interior()) {
+				// Sibling parachain
+				(1, Some(Parachain(id))) => Some((
+					MultiLocation::new(1, X1(Parachain(*id))),
+					MultiLocation::new(0, dest.interior().clone().split_first().0),
```

Comment:
Do you mean `split_first().1`?

Reply:
Because I think `split_first().0` should be `X1(Parachain(id))` in this case. It's not that useful?

Reply:
No, the first item represents the remaining suffix. See [here](https://github.com/paritytech/polkadot/blob/b4cf84ddf8cf5c7ecfaf2cb906f502c658fe7694/xcm/src/v1/multilocation.rs#L558)